admin/spam/views.py

```diff
@@ -1,9 +1,119 @@
-from django.http import HttpResponse
+import operator
 
+from django.shortcuts import render, render_to_response
+from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
+from django.contrib.auth.decorators import login_required
+from django.core.mail import send_mail
+from django.views.generic import FormView
+from django.core.urlresolvers import reverse
 
+from modularodm import Q
+from website.project.model import Comment
+from website.settings import SUPPORT_EMAIL
+
+from .serializers import serialize_comment
+from .forms import EmailForm
+
+
+class EmailFormView(FormView):
+
+    form_class = EmailForm
+    template_name = "spam/email.html"
+    spam_id = None
+    page = 1
+
+    def get(self, request, *args, **kwargs):
+        spam_id = kwargs.get('spam_id', None)
+        self.spam_id = spam_id
+        self.page = request.GET.get('page', 1)
+        try:
+            spam = serialize_comment(Comment.load(spam_id))
+        except:
+            pass
+        self.set_initial(spam)
+        form = self.get_form()
+        context = {
+            'comment': spam,
+            'page_number': request.GET.get('page', 1),
+            'form': form
+        }
+        return render(request, self.template_name, context)
+
+    def post(self, request, *args, **kwargs):
+        spam_id = kwargs.get('spam_id', None)
+        self.spam_id = spam_id
+        self.page = request.GET.get('page', 1)
+        try:
+            spam = serialize_comment(Comment.load(spam_id))
+        except:
+            pass
+        self.set_initial(spam)
+        form = self.get_form()
+        if form.is_valid():
+            return self.form_valid(form)
+        else:
+            return self.form_invalid(form)
+
+    def set_initial(self, spam):
```

Comment:
Would it make more sense to override `FormMixin#get_initial` instead of using this setter?